### ISSUE TITLE
ChaCha20/Poly1305

### ISSUE THREAD

**Issue by yaronf** [yaronf/I-D#201]
Should probably mention, and add to recommended ciphersuites, but let's consult the UTA WG and CFRG

DONE: Peter check on implementation in browsers (esp. mobile)
* Usage in Firefox is around 0.5%
* Apparently ChaCha20/Poly1305 is preferred by Chrome on Android
* And of course it's a SHOULD in RFC 8446, too.

**Comment by yaronf** [comment]
Add one ciphersuite for TLS 1.2 that includes chacha-poly.

**Comment by stpeter** [comment]
I've sent an email to some Google folks about usage on Chrome and will report back.

**Comment by thomas-fossati** [comment]
> Add one ciphersuite for TLS 1.2 that includes chacha-poly.

To keep it aligned with the list in Section 4.2, it should probably be RFC7905's `TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256`.

Although, this raises a (maybe silly) question: what is the rationale for not recommending the ECDSA variants too -- in parallel with RSA?

**Comment by yaronf** [comment]
Because when we wrote the RFC, the thinking was that real-life certs are exclusively RSA. (And IoT does not count as real life...)

If you think this is not true any more, we need to open a separate issue for that.

**Comment by thomas-fossati** [comment]
> Because when we wrote the RFC, the thinking was that real-life certs are exclusively RSA. (And IoT does not count as real life...)

LOL

> If you think this is not true any more, we need to open a separate issue for that.

I guess this might change a bit the context: https://community.letsencrypt.org/t/ecdsa-availability-in-production-environment/150679

**Comment by yaronf** [comment]
Maybe. Asking seriously: what's to motivate a web site owner (or mail server owner, whatever) to switch to an ECDSA cert?

**Comment by thomas-fossati** [comment]
> Maybe. Asking seriously: what's to motivate a web site owner (or mail server owner, whatever) to switch to an ECDSA cert?

PRO
* sending less bytes
* savings in processing (5-10x signatures compared to RSA -- the client pays the cost on signature verification)

CON
* If compatibility with very old systems (e.g., if you have to support *very* old Android gear), then you might want to think twice because the cost of dealing with a dual ECDSA/RSA chain may not be trivial

> Bear in mind that we are not speaking about greenfield deployments here. Those are people who do support TLS 1.2, i.e. older clients.

Sure, in addition to LetsEncrypt having now an ECDSA hierarchy up & running in production (and Certbot support since last December),  if you look at Cloudflare's supported ciphersuites [https://developers.cloudflare.com/ssl/ssl-tls/cipher-suites] you’ll see all the ECDSA's have a green flag for 1.2

**Comment by yaronf** [comment]
Moved the ECDSA discussion to #218, to focus on chacha/poly here.

**Comment by thomas-fossati** [comment]
An old graph (2015) from CF showing the uptake of ChaCha:

![](https://blog.cloudflare.com/content/images/2015/02/illustration-ciphersuite-percentage-ssl-week.png)

**Comment by thomas-fossati** [comment]
A more recent (2019) report from F5's vantage point sees ChaCha at ~3%.  See fig. 5 of https://www.f5.com/content/dam/f5-labs-v2/article/pdfs/F5Labs-2019-TLS-Telemetry-Report-Summary.pdf

**Comment by yaronf** [comment]
It's not clear cut, but on consideration, I think we should *not* add a chcha/poly cipher suite. It's use is not widespread enough to make it important for interoperability, and neither is it likely to grow very much (what with ARM supporting AES). Also, the main goal of the document is security, not performance, so I don't think this is in scope for bcp195.

**Comment by thomas-fossati** [comment]
I understand your point, but I am a bit nervous about the "AES monoculture" side of the current recommendation.  while I have no reason to think AES is likely to fail catastrophically I guess providing a minimal amount of diversity in the ciphersuite choice makes the recommendation more robust?

**Comment by yaronf** [comment]
https://datatracker.ietf.org/doc/html/draft-mcgrew-standby-cipher-00

@stpeter ?

**Comment by stpeter** [comment]
I tend to agree with Thomas that (all other things being equal) ciphersuite diversity is a good thing. Because TLS_CHACHA20_POLY1305_SHA256 [RFC8439] is RECOMMENDED in RFC 8446, it makes sense that we would recommend it in 7525bis as well. @yaronf is your concern specifically related to recommending this ciphersuite for TLS 1.2?

**Comment by yaronf** [comment]
Yes. Right now we are only making specific recommendations for 1.2 (so far, unchanged from the RFC), while referring readers to RFC 8446 for TLS 1.3 cipher suites. So the discussion is only about 1.2.

**Comment by stpeter** [comment]
Let's discuss in our next call. I lean slightly toward recommending it, but I might want to gather more data.

**Comment by stpeter** [comment]
Clearly we have some difference of opinion among the authors. I wonder if we should bring this to the UTA list for discussion.

**Comment by thomas-fossati** [comment]
> Clearly we have some difference of opinion among the authors. I wonder if we should bring this to the UTA list for discussion.

sounds to me like a very sensible plan

**Comment by yaronf** [comment]
Mail sent to UTA.

**Comment by yaronf** [comment]
Closing.
